### PR TITLE
Update Gambit and Gerbil

### PR DESCRIPTION
**Pull request by fare**
###### Motivation for this change

Upstream update + make Gambit stable buildable with recent Nix.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/gambit/default.nix [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 callPackage ./build.nix {
   version = "4.8.9";
+  git-version = "v${version}";
```
Comment (xeji) [inline pkgs/development/compilers/gambit/default.nix]: Eval error because this isn't a recursive set. Try using `callPackage ./build.nix rec {` to make it recursive.

Comment (fare) [reply pkgs/development/compilers/gambit/default.nix]: Oops. Thanks a lot! Fixed.

File: pkgs/development/compilers/gambit/build.nix [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{ stdenv, git, openssl, autoconf, pkgs, makeStaticLibraries, version, SRC }:
+{ stdenv, git, openssl, autoconf, pkgs, makeStaticLibraries, version, SRC, git-version }:
```
Comment (xeji) [inline pkgs/development/compilers/gambit/build.nix]: The `git-version` argument isn't used anywhere. Why did you add it?

File: pkgs/development/compilers/gambit/default.nix [hunk 3]
```diff
@@ -1,7 +1,8 @@
 { callPackage, fetchurl }:
 
-callPackage ./build.nix {
+callPackage ./build.nix rec {
   version = "4.8.9";
+  git-version = "v${version}";
```
Comment (xeji) [inline pkgs/development/compilers/gambit/default.nix]: Defining `git-version` only makes sense if it is actually used by `build.nix`.

Comment (fare) [reply pkgs/development/compilers/gambit/default.nix]: Well,

1- I include it by symmetry with gerbil, that actually uses it. I actually intended to use it in gambit in the very same way, but that requires changes to the gambit build system similar to those done in Gerbil, that haven't made it on top of my todo list yet.

2- It's somewhat useful for me to track the progress of gambit unstable, and needed by symmetry in the stable build, too.

3- If you object to it, I can make it a comment instead, until I get to modifying the gambit build system (and get the changes upstreamed).

Comment (xeji) [reply pkgs/development/compilers/gambit/default.nix]: Thank you for explaining. While I understand your intentions, unused code tends to confuse people and make packages harder to maintain - you might not be the only one modifying these  in the future. So please remove the unused code but feel free to add your comments.

Comment (fare) [reply pkgs/development/compilers/gambit/default.nix]: No problem. I understand your concerns. Here is an updated version where only the unstable build has the git-version, and only in a comment.

**Comment by xeji** [comment]
@GrahamcOfBorg build gambit-unstable gerbil-unstable

**Comment by fare** [comment]
It does take a lot of time to build it — tens of minutes on my machine from last year.

Also, I see you use clang. I believe gambit currently works better with GCC. How do I force the use of GCC for Gambit and Gerbil?

**Comment by fare** [comment]
Excerpt from an email last May by Marc Feeley, author of Gambit, on the gambit-list:
```
the build time of a “make -j8” on my 4 core laptop is over 10x slower when using CLANG.  As for speed of the generated code, CLANG produces code that is about 3x slower than GCC.
```

**Comment by xeji** [comment]
clang is the default compiler on darwin (MacOS), while x86_64-linux and aar64_linux use gcc7.
Not sure we support gcc on darwin.

**Comment by fare** [comment]
It makes total sense as a default on Darwin. But can I somehow override the default?

**Comment by xeji** [comment]
Timeouts on ofborg are not uncommon since ofborg uses community-provided build machines that don't guarantee good performance. These packages usually build on hydra (higher time limit and better performance), so it should be fine without any changes.

To force gcc on darwin, you can add `stdenv = gccStdenv;` for the package in `all-packages.nix`,  see other examples there. No guarantees, it may time out on ofborg as well. But feel free to give it a try.

**Comment by fare** [comment]
Thanks, xeji. I added a commit that forces Gambit and Gerbil to use GCC rather than Clang.

**Comment by fare** [comment]
NB: I checked that these changes are NOPs on my Linux amd64 NixOS box.

**Comment by xeji** [comment]
@GrahamcOfBorg build gambit-unstable gerbil-unstable

**Comment by fare** [comment]
@vyzo can you give a look at the darwin failure? If not, who is building Gerbil on Darwin?

**Comment by fare** [comment]
This looks like a recent upstream bug in the new kqueue support.

@grahamc What version of macOS is the Nix builder running? How do I tell?

**Comment by vyzo** [comment]
The darwin failure should be fixed by https://github.com/vyzo/gerbil/pull/118

**Comment by fare** [comment]
@GrahamcOfBorg build gambit-unstable gerbil-unstable

I updated this PR to include @vyzo's latest upstream fixes.

**Comment by 7c6f434c** [comment]
@GrahamcOfBorg build gambit-unstable gerbil-unstable

**Comment by 7c6f434c** [comment]
Let me just say we don't block Chromium build attempts on ofBorg (obviously, they time out always and everywhere). No tooling is perfect
